extensions/data-workspace/src/common/interfaces.ts

```diff
@@ -81,6 +81,14 @@ export interface IWorkspaceService {
 	 */
 	createProject(name: string, location: vscode.Uri, projectTypeId: string, workspaceFile?: vscode.Uri): Promise<vscode.Uri>;
 
+	/**
+	 * Clones git repository and adds projects to workspace
+	 * @param url The url to clone from
+	 * @param localClonePath local path to clone repository to
+	 * @param workspaceFile workspace file to add the projects to
+	 */
+	gitCloneProject(url: string, localClonePath: string, workspaceFile: vscode.Uri): Promise<void>;
```

Comment:
nit: will this allow only single project cloning at a time? if more than that should we name project(s) or workspace or euch?

Reply:
it will add all the projects in the cloned repo to the workspace for now. https://github.com/microsoft/azuredatastudio/blob/141912d26a42c05b20e05593b479ccb0cf93343e/extensions/data-workspace/src/services/workspaceService.ts#L289-L291